continuous-integration/reference-architectures/README.md

```diff
@@ -1,3 +1,181 @@
+
+### An approach to best practices
+Routine merging of components into a functional software product and publishing compiled binaries 
+is the core of CI. It demands code be committed collaboratively and requires products from different 
+development systems to work together. It tests software builds through a predictable process of known 
+touchpoints that identify specific areas of interoperability and breakage. Compilation results are 
+shared in a centralized system of logging and reporting. These approaches promote many software 
+development best practices, but in particular [reduce risk](https://martinfowler.com/articles/continuousIntegration.html#BenefitsOfContinuousIntegration):
+- [Single Source of Truth](https://martinfowler.com/articles/continuousIntegration.html#MaintainASingleSourceRepository.): 
+All components and configurations must be bundled to prove compiles function, easing deployment of 
+compiled binaries.
+- [Fail Fast/Fix Fast](https://martinfowler.com/articles/continuousIntegration.html#FixBrokenBuildsImmediately): 
+Know the limitations of compiled integrations and receive immediate notifications about failures.
+- [Visibility/Open Success](https://martinfowler.com/articles/continuousIntegration.html#EveryoneCanSeeWhatsHappening): 
+The state of the product is reported constantly on CI services.
+- [Testing Availability](https://martinfowler.com/articles/continuousIntegration.html#AutomateDeployment): 
+Automation allows repeatable deployment to test servers, elevating functional testing on par 
+with release.
+
+### Pipeline: A series of systems executing builds
+A [CI pipeline](https://www.redhat.com/en/topics/devops/what-cicd-pipeline) is a series of automated 
+processes, systems and triggers that execute in order to push software from raw code to a compiled 
+product that may be deployed.  
+```mermaid
+flowchart LR
+    subgraph P["<b>Pipeline</b>"]
+        subgraph DS[Development System]
+            P1
+            P2
+            P1[Commit]-->P2[Validate]
+        end
+        subgraph CIS[Continuous Integration System]
+            P3
+            P4
+            P5
+            P6
+            P7
+            P3[Compile]-->P4[Test]
+            P4-.->P5[Package]
+            P4-->P6[Report]
+            P5-->P7[Publish]
+        end
+        subgraph CDS[Continuous Deployment System]
+            P8
+            P9
+            P8[Orchestrate]-->P9[Release]
+            P9-.->P8
+        end
+        
+        DS --> |trigger| CIS
+        CIS --> |trigger| CDS
+    end
+    style P fill:#A7C7E7,stroke:#666,stroke-width:4px
+    style DS fill:#E5E4E2,stroke:#333,stroke-width:2px
+    style CIS fill:#E5E4E2,stroke:#333,stroke-width:2px
+    style CDS fill:#E5E4E2,stroke:#333,stroke-width:2px,stroke-dasharray: 5 5
+```
+The CI pipeline is composed of several systems linked together through a series of automation 
+triggers: 
+1. _Development System_  
```

Comment:
Is there a way to add the numbering you have here to your diagram? That would make the mappings much clearer.

Reply:
Added. This will show up in the next commit.